src/dc_sqlite3.rs

```diff
@@ -13,43 +13,67 @@ const DC_OPEN_READONLY: usize = 0x01;
 
 /// A simple wrapper around the underlying Sqlite3 object.
 #[repr(C)]
-pub struct dc_sqlite3_t {
-    pub cobj: *mut sqlite3,
+pub struct SQLite {
+    pub cobj: std::sync::RwLock<*mut sqlite3>,
 }
 
-pub fn dc_sqlite3_new() -> dc_sqlite3_t {
-    dc_sqlite3_t {
-        cobj: std::ptr::null_mut(),
+impl SQLite {
+    pub fn new() -> SQLite {
+        SQLite {
+            cobj: std::sync::RwLock::new(std::ptr::null_mut()),
+        }
     }
-}
 
-pub unsafe fn dc_sqlite3_unref(context: &Context, sql: &mut dc_sqlite3_t) {
-    if !sql.cobj.is_null() {
-        dc_sqlite3_close(context, sql);
+    pub fn is_open(&self) -> bool {
+        !self.cobj.read().unwrap().is_null()
```

Comment:
i do not understand this line - where did the close() go to?

Reply:
the dc_sqlite3_unref() function is completely gone.  is_open() should not close the connection... and replaces the dc_sqlite3_is_open() call (which is also gone).  The diff is just confusing putting the wrong things next to each other.